dev_scripts/download_pubchem.py

```diff
@@ -0,0 +1,148 @@
+"""
+This module implements a script for extracting molecule names, SMILES and xyz
+files from the Pubchem database.
+
+It uses multiprocessing to process many files simultaneously.
+
+WARNING: While in progress the disk usage maybe fairly large (~100 GB).
+
+Writes to MongograntStore. The data will be inserted into the "collection"
+collection in the "mp_pubchem" database in the mongo database.
+"""
+
+import ftplib
+import pebble
+import glob
+import os
+
+import pybel
+from pebble import ProcessExpired
+
+from tqdm import tqdm
+from concurrent.futures import TimeoutError
+from maggma.advanced_stores import MongograntStore
+
+tmp_dir = "tmp"
+
+keys = ['PUBCHEM_OPENEYE_CAN_SMILES', 'PUBCHEM_OPENEYE_ISO_SMILES',
+        'PUBCHEM_IUPAC_CAS_NAME', 'PUBCHEM_IUPAC_NAME',
+        'PUBCHEM_IUPAC_TRADITIONAL_NAME', 'PUBCHEM_IUPAC_SYSTEMATIC_NAME',
+        'PUBCHEM_IUPAC_OPENEYE_NAME']
+
+key_map = {'PUBCHEM_OPENEYE_CAN_SMILES': 'smiles_can',
+           'PUBCHEM_OPENEYE_ISO_SMILES': 'smiles_iso',
+           'PUBCHEM_IUPAC_CAS_NAME': 'name_cas',
+           'PUBCHEM_IUPAC_NAME': 'name_iupac',
+           'PUBCHEM_IUPAC_TRADITIONAL_NAME': 'name_traditional',
+           'PUBCHEM_IUPAC_SYSTEMATIC_NAME': 'name_systematic',
+           'PUBCHEM_IUPAC_OPENEYE_NAME': 'name_openeye'}
+
+
+def process_sdf_file(filename):
+    mp_pubchem = MongograntStore("rw:knowhere.lbl.gov/mp_pubchem", "mp_pubchem",
```

Comment:
This repository was set up by @shyamd as he is interested in using the data for MP.

The issue is the pubchem data is several hundred GBs, so running this script is not part of routine work on robocrystallographer.

I think it would make sense to transition at least the `download_pubchem` script outside of Robocrystallographer, and add information to contact Shyam if you need access to the database to regenerate the local mineral db.

Reply:
I've added clarification on what the maggma store is and how to reconfigure the script to write to a local MongoDB database.